x/lightclient/types/state.go

```diff
@@ -0,0 +1,90 @@
+package types
+
+import (
+	"bytes"
+	"time"
+
+	errorsmod "cosmossdk.io/errors"
+	abci "github.com/cometbft/cometbft/abci/types"
+	tmprotocrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	cmttypes "github.com/cometbft/cometbft/types"
+
+	ibcclienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+// CheckCompatibility checks if the IBC state and Rollapp state are compatible
+func CheckCompatibility(ibcState IBCState, raState RollappState) error {
+	// Check if block descriptor state root matches IBC block header app hash
+	if !bytes.Equal(ibcState.Root, raState.BlockDescriptor.StateRoot) {
+		return errorsmod.Wrap(ibcclienttypes.ErrInvalidConsensus, "block descriptor state root does not match tendermint header app hash")
+	}
+	// Check if the validator pubkey matches the sequencer pubkey
+	if len(ibcState.Validator) > 0 && !bytes.Equal(ibcState.Validator, raState.BlockSequencer) {
+		return errorsmod.Wrap(ibcclienttypes.ErrInvalidConsensus, "validator does not match the sequencer")
+	}
+	if len(raState.NextBlockSequencer) == 0 {
+		return ErrNextBlockDescriptorMissing
+	}
```

Comment:
This can happen during the `AfterStateUpdate` hook, the ibc consensus state exists for the last BD in the stateinfo. and the h+1 BD would be part of the next stateinfo, which isnt available yet.